tpu-provisioner/internal/cloud/gke.go

```diff
@@ -311,6 +326,13 @@ func sumTPURequests(p *corev1.Pod) (int, error) {
 }
 
 func podToNodePoolName(p *corev1.Pod, prefix, suffix string) string {
+	// Use the UID of the Pod as the unique identifier for the node pool.
```

Comment:
Nit: I think this should say we try to use the UID of the Job which owns the pod first, but only fall back to the pod UID if owner UID is unavailable, right?

Reply:
Good point